### DISCUSSION TITLE
plot of vpc

### DISCUSSION THREAD

**Discussion by Mannny-PK** [LAPKB/Pmetrics#90] [Q&A]
Hello, I would like to ask a question: How to successfully export the plot of vpc?
When I run the following code it prompts an error
![屏幕截图 2022-08-17 234252](https://user-images.githubusercontent.com/109020077/185271519-2ebe688c-5023-40b2-9020-e85445fd946f.png)
Then I tried the makevalid function, but it still failed.
![屏幕截图 2022-08-17 234441](https://user-images.githubusercontent.com/109020077/185271635-3eafc079-bfb5-40b3-874e-fda0a0d1b40f.png)

**Comment by yoannc40** [comment]
Hello,

I have the same kind of problem with a different error message:
![image](https://user-images.githubusercontent.com/8640718/188648572-a774d191-f8c1-4d10-b49f-f52341d2025d.png)

So I used "makeValid" and then I have this kind of error:
![image](https://user-images.githubusercontent.com/8640718/188649179-82385012-35b6-4806-b657-e5c46cb9ff24.png)
![image](https://user-images.githubusercontent.com/8640718/188649327-24d90f2a-54f0-414f-8d5c-7cfdc5b1a579.png)

does anyone have a solution?

Kind regards,

Yoann

**Comment by yoannc40** [comment]
@Siel, Any idea for the reason of this kind of error message: concerning the function makeValid()? So we can't use anymore makeNPDE function?

Best regards,
Yoann

**Comment by mnneely** [comment]
That is correct, makeNPDE is deprecated.  I suspect the error you are seeing is if you have EVID=4 events in your data. That should be corrected in the beta Version2 branch. However, that branch makes a lot of changes to Pmetrics as we move to a more modern framework. We are doing our best to maintain backwards compatibility with versions prior to Version 2, but we can’t guarantee that it will be seemless and everyone’s scripts will work fully. However, we remain committed to working through issues.

If you want to try the beta Version2, use devtools::install_github(“LAPKB/Pmetrics”, ref = “Version2”).

From: yoannc40 ***@***.***>
Date: Friday, September 9, 2022 at 5:29 AM
To: LAPKB/Pmetrics ***@***.***>
Cc: Subscribed ***@***.***>
Subject: Re: [LAPKB/Pmetrics] plot of vpc (Discussion #90)

@Siel<https://urldefense.com/v3/__https:/github.com/Siel__;!!LIr3w8kk_Xxm!p8v1ANqdv1LU6qP3EhZQAmqJrHNzrOAKrwPzxbZwjdd5kE3ZdSw8lpC-vcGFZiVbZ68nvv4V7TZcpKYoST4iVTI$>, Any idea for the reason of this kind of error message: concerning the function makeValid()? So we can't use anymore makeNPDE function?

Best regards,
Yoann

**Reply by yoannc40** [reply]
Dear Michael,

Thanks for the answer and thanks for your work.

I have no EVID4 in my data and I use version 2.
I can't use version 1.9.7 since it doesn't work with my UC configuration.

No other ideas?

Best regards,
Yoann

**Reply by yoannc40** [reply]
I have this kind of error with the Example.

Thx

**Reply by mnneely** [reply]
With the current Version2, I believe at least that the example is now working properly. I appreciate your patience with this. It is quite a tricky function to program. If you'd like to reinstall version2 via install_github() and let me know how it goes, I can continue to debug as needed.  Thanks.
